MonoGame.Framework/Graphics/SpriteBatchItem.cs

```diff
@@ -19,70 +19,77 @@ public SpriteBatchItem ()
             vertexBR = new VertexPositionColorTexture();            
 		}
 		
-		public void Set ( float x, float y, float w, float h, Color color, Vector2 texCoordTL, Vector2 texCoordBR )
+		public void Set ( float x, float y, float w, float h, Color color, Vector2 texCoordTL, Vector2 texCoordBR, float depth )
 		{
 			vertexTL.Position.X = x;
```

Comment:
Most cycles in the SpriteBatcher loop are spent to copy 4 `TextureColorTexture` vertices from `SpriteBatchItem` to the `vertexArray`. It 's about 96 bytes or so total.
https://github.com/nkast/MonoGame/blob/6121e4b6d13a835206d00a951587e12d06baec6d/MonoGame.Framework/Graphics/SpriteBatcher.cs#L234-L237 
By moving the calculation, the `Set` method will only have to copy around 40 bytes into the smaller `SpriteBatchItem`. That's where my bets are.
SpriteBatchItem are currently  The `Sort` moves object references (pointers) around which are the size of a register (32/64bit), but then, there are also cache misses when it access the `SortKey`, a smaller `SpriteBatchItem` will definitely help but I can't speculate on or measure  those kind of things, only profiling will show if it's significant. Allocating all items was definitely good, unless the CLR decide to move items around... Also Sorting will shuffle references around over time. The item at array[0] will not be the first item we allocated.




Another idea to remove those 4 lines from the loop was to have `Set` write directly into the `VertexArray`! The SpriteBatchItem will hold a reference to the buffer and an incrementing index. This is straightforward for deferred mode. In case of other type of sprite sorting we can use the BatchItem index to recreate a secondary indexBuffer. That's one memory read (index) and 6 writes of `short`. Still haven't work out the details (resizing, etc).